homeassistant/helpers/state.py

```diff
@@ -209,11 +210,20 @@ def state_as_number(state):
 
     Raises ValueError if this is not possible.
     """
-    if state.state in (STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON,
-                       STATE_OPEN, STATE_HOME):
+    on_states = {STATE_ON, STATE_LOCKED, STATE_ABOVE_HORIZON, STATE_OPEN,
+                 STATE_HOME, STATE_HEAT, STATE_COOL}
+    off_states = {STATE_OFF, STATE_UNLOCKED, STATE_UNKNOWN,
+                  STATE_BELOW_HORIZON, STATE_CLOSED, STATE_NOT_HOME,
+                  STATE_IDLE}
+
+    states = {state.state}
+    if isinstance(state.state, str):
+        # Also try to match lowercase version of the state string.
+        states.add(state.state.lower())
```

Comment:
We have a standardized set of states based on predefined constants (as defined above). Integrations are not supposed to return other states than our predefined states. Instead of allowing broken integrations, we should make sure that integrations never return custom states.

Please remove this code.

Reply:
Thanks for giving me some guidance (I am new to home-assistant), very helpful.

The commit now simply changes the helper to look at climate states as well, and I will try to submit a separate PR that will improve zwave to actually use predefined states instead of custom strings.